### PR TITLE
Reorganizing the files. 

### PR DESCRIPTION
**Pull request by kegl**
ipython notebook and user_test_submission comes to main (root). 
We add data/public for public data sets, both notebook and user test reads from there.
We add submissions/<submission_name> for example submissions.
We will require at least submissions/starting_kit/*.py which user_test_submission will execute, but organizer can provide other example submissions, too.

### CODE REVIEW

File: user_test_submission.py [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
-        clf.fit(X_train_array, y_train_array)
-        y_proba = clf.predict_proba(X_test_array)
-
-        print('roc auc score = ', roc_auc_score(y_test_array, y_proba[:, 1]))
+import pandas as pd
+from sklearn.model_selection import StratifiedShuffleSplit
+from sklearn.metrics import roc_auc_score
+
+import submissions.starting_kit.feature_extractor as feature_extractor
+import submissions.starting_kit.classifier as classifier
```
Comment (agramfort) [inline user_test_submission.py]: from submissions.starting_kit import classifier, feature_extractor

File: user_test_submission.py [hunk 2]
```diff
@@ -1,43 +1,43 @@
...
+
+
+train_filename = 'data/public/public_train.csv'
+
+
+def read_data(filename):
+    data = pd.read_csv(filename)
+    y_array = data['Survived'].values
+    X_df = data.drop(['Survived', 'PassengerId'], axis=1)
```
Comment (agramfort) [inline user_test_submission.py:14]: inplace=True

File: .travis.yml [hunk 3]
```diff
@@ -11,8 +11,8 @@ install:
     - conda install --yes --quiet numpy scipy matplotlib scikit-learn pandas
     - pip install -q flake8
 script:
-    - cd starting_kit
-    - flake8 *.py
+    - flake8 user_test_submission.py
+    - flake8 submissions/*/*.py
```
Comment (agramfort) [inline .travis.yml]: you can do it on one line

flake8 *.py submissions/*/*.py

**Comment by agramfort** [comment]
besides LGTM

File: .travis.yml (hunk 3 above)
Comment (kegl) [reply .travis.yml]: It's not 
flake8 *.py submissions//*.py
?

Comment (kegl) [reply .travis.yml]: The star doesn't show up in the rendered text :)

File: user_test_submission.py (hunk 2 above)
Comment (kegl) [reply user_test_submission.py:14]: Doesn't work:

silver6:titanic kegl$ python user_test_submission.py 
Reading file ...
Training file ...
Traceback (most recent call last):
  File "user_test_submission.py", line 26, in <module>
    for train_is, test_is in skf.split(X_df, y_array):
  File "/Users/kegl/anaconda/lib/python2.7/site-packages/sklearn/model_selection/_split.py", line 953, in split
    for train, test in self._iter_indices(X, y, groups):
  File "/Users/kegl/anaconda/lib/python2.7/site-packages/sklearn/model_selection/_split.py", line 1250, in _iter_indices
    n_samples = _num_samples(X)
  File "/Users/kegl/anaconda/lib/python2.7/site-packages/sklearn/utils/validation.py", line 122, in _num_samples
    type(x))
TypeError: Expected sequence or array-like, got <type 'NoneType'>

**Comment by agramfort** [comment]
ok now we have travis visible on the PRs !

**Comment by agramfort** [comment]
ok forget it then

**Comment by agramfort** [comment]
travis is happy i am happy !

**Comment by agramfort** [comment]
thx @kegl

**Pull request merged**